blog/_posts/2024-10-15-scala-3.5.0-released.md

```diff
@@ -0,0 +1,119 @@
+
+As os-lib is a part of the default Scala toolkit, we can replace the `//> using dep com.lihaoyi::os-lib:0.10.3` line with `//> using toolkit default`. Furthermore, we can add  `#!/usr/bin/env -S scala shebang` at the top of the file. Then, after setting the permissions, we can treat `biggerThan.scala` as any executable script. Invoking `./biggerThan.scala ../my-directory 10` will incrementally compile the file and then run it.
+
+To learn the full scope of the new capabilities, read about Scala CLI [commands](https://scala-cli.virtuslab.org/docs/commands/basics/) and [using-directives](https://scala-cli.virtuslab.org/docs/reference/directives) or glance at [the cookbook](https://scala-cli.virtuslab.org/docs/cookbooks/intro).
+
+Merging Scala CLI into the compiler doesn't change the behavior of popular build tools, such as sbt or Mill. It will, however, allow for maintaining and publishing single-module multiplatform libraries without any build tool.
+
+## What's new in 3.5.0?
+
+### Best Effort Compilation
+
+When using Metals, the IDE functions work great as long as the code passes the compilation. However, once the user starts changing the code, the support drops in quality. This gets worse with a larger number of compilation errors and subsequent modifications. In 3.5.0, we have fixed this problem. We introduced a new mode of compilation called Best Effort Compilation. When enabled, the compiler will output BETASTy files for not compiling code. It can be used by tooling to provide autocompletion and other IDE features.
+
+To enable the use of BETASTy files in Metals, start the language server with `-Dmetals.enable-best-effort=true` or put that into `metals.serverProperties` setting in VS Code.
+
+### Support for Pipelined Builds
+
+Scala 3.5.0 supports pipelined compilation. It can be enabled by setting `ThisBuild/usePipelining := true` in sbt build definition. This can result in significant speedups in compilation time for multi-module projects.
+
+You can learn more about how the pipelined compilation works and what benefits you can expect from [the talk by Jamie Thompson](https://www.youtube.com/watch?v=1uuFxEAiPuQ&t=1473s).
+
+### `var` in refinements
+
+Until now, only types, `val`s, and `def`s were allowed in type refinements. In 3.5.0, `var`s can also be declared.
+
+```scala
+type A = { var number: Int }
+```
+
+is now legal and equivalent to
+
+```scala
+type A = { def number: Int, def number_=($number: Int): Unit }
+```
+
+This change can simplify libraries based on metaprogramming using type refinements.
+
+### Support for binary integer literals
+
+Integer literals can now start with `0b` or `0B`. They will be interpreted as numbers in base 2.
+
+```scala
+assert(0B1 == 1)
+assert(0B10 == 2)
+assert(0B_1000_0010 == 130)
+assert(0B_1000_0010 == 0x82)
+```
+
+## Work on a better scheme for given prioritization
```

Comment:
This section may seem longer than necessary, but we really want people to test the new given prioritization scheme.